### ISSUE TITLE
Fragile compiler detection test

### ISSUE THREAD

**Issue by kali** [flame/blis#225]
What is the rationale for using this test (nothing on stderr) for compiler sanity check ?

https://github.com/flame/blis/blob/2000cdff59272974438e88e0e82d8e1a32710325/configure#L846

clang does something weird on macOS. When invoke as "gcc --version", it spews some configuration details on stderr.

```
% gcc --version                                                                     dev
Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
% clang --version                                                                  dev
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
% cc --version                                                                     dev
Apple LLVM version 9.1.0 (clang-902.0.39.2)
Target: x86_64-apple-darwin17.6.0
Thread model: posix
InstalledDir: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin
```

It's not a blocker, I only encountered it because my Rust binding build helpers set CC=gcc, so I can work around it. Just wondering.

**Comment by fgvanzee** [comment]
@kali It's not clear to me what problem you are reporting. Perhaps you could rephrase and clarify?

Your question, however, led me to study `select_cc()` more closely. I fixed several bugs (3f387ca) that typically did not manifest but that eventually, given enough time, would bite someone. So, thanks for that. :)

**Comment by kali** [comment]
Yeah, I could have been more explicit. I apologize. I meant that an empty stderr was not a good enough criteria as clang invoked as gcc is actually printing the `Configured with: --prefix=/Applications/Xcode.app/Contents/Developer/usr --with-gxx-include-dir=/usr/include/c++/4.2.1` line on stderr.

Switching to a return code check is most certainly the right option. (except maybe for windows ?)

I can not check this right now, but you've probably fixed it. Moving to close, at least for now.

Thanks for deciphering this more cryptic than necessary bug report and actually doing something about it :)
